odoo/api.py

```diff
@@ -539,37 +540,63 @@ def user(self):
         """ return the current user (as an instance) """
         return self(su=True)['res.users'].browse(self.uid)
 
+    def _has_valid_companies(self):
```

Comment:
Make it more a "filter" function that works on a parameter and returns a filtered result
(or inline it in the callers)
Having it share an implicit parameter with callers (via context) is confusing.